columnflow/tasks/framework/remote.py

```diff
@@ -227,38 +228,41 @@ class BundleCMSSWSandbox(AnalysisTask, law.cms.BundleCMSSW, law.tasks.TransferLo
     )
 
     def __init__(self, *args, **kwargs):
-        # cached bash sandbox that wraps the cmssw environment
-        self._cmssw_sandbox = None
-
         super().__init__(*args, **kwargs)
 
+        # get the name and install path of the sandbox
+        from cf_sandbox_file_hash import create_sandbox_file_hash
+        sandbox_file = os.path.expandvars(os.path.expanduser(self.sandbox_file))
+        self.sandbox_file_hash = create_sandbox_file_hash(sandbox_file)
+        self.cmssw_env_name = os.path.splitext(os.path.basename(sandbox_file))[0]
+        self.cmssw_env_name_hashed = f"{self.cmssw_env_name}_{self.sandbox_file_hash}"
+
     def requires(self):
         return self.reqs.BuildBashSandbox.req(self)
 
     def get_cmssw_path(self):
         # invoking .env will already trigger building the sandbox
         return self.requires().sandbox_inst.env["CMSSW_BASE"]
 
-    def get_file_pattern(self):
-        path = os.path.expandvars(os.path.expanduser(self.single_output().path))
-        return self.get_replicated_path(path, i=None if self.replicas <= 0 else r"[^\.]+")
-
     def single_output(self):
         cmssw_path = os.path.basename(self.get_cmssw_path())
-        return self.target(f"{cmssw_path}.{self.checksum}.tgz")
+        return self.target(f"{self.cmssw_env_name_hashed}_{cmssw_path}.{self.checksum}.tgz")
```

Comment:
Given that the tarball name now includes the hash, could the `cmsw_path` part actually be dropped here?

Reply:
The installation directory for cmssw is e.g. `cmssw_default_b6e834cc/CMSSW_12_6_2`, driven by the way `cmsrel` checks out releases.Aand it could be that the same sandbox file (same hash) switches the CMSSW version. This would also need to be covered by the target path above (not 100% cleam, but cmssw compatible) :+1:

Reply:
Ah, I see, thanks for the explanation! Then of course it makes sense to keep the CMSSW version as part of the tarball name 👍